src/Roslyn.Suppressions/Roslyn.Suppressions/InProcess/SuppressedIssuesFileSynchronizer.cs

```diff
@@ -64,36 +70,55 @@ internal sealed class SuppressedIssuesFileSynchronizer : ISuppressedIssuesFileSy
             ISonarQubeIssuesProvider suppressedIssuesProvider,
             ISuppressedIssuesFileStorage suppressedIssuesFileStorage,
             IActiveSolutionBoundTracker activeSolutionBoundTracker,
+            ILogger logger,
             IThreadHandling threadHandling)
         {
             this.suppressedIssuesMonitor = suppressedIssuesMonitor;
             this.suppressedIssuesProvider = suppressedIssuesProvider;
             this.suppressedIssuesFileStorage = suppressedIssuesFileStorage;
             this.activeSolutionBoundTracker = activeSolutionBoundTracker;
+            this.logger = logger;
             this.threadHandling = threadHandling;
 
             suppressedIssuesMonitor.SuppressionsUpdateRequested += SuppressedIssuesMonitor_SuppressionsUpdateRequested;
         }
 
         private void SuppressedIssuesMonitor_SuppressionsUpdateRequested(object sender, EventArgs e)
         {
-            UpdateFileStorage();
+            // Called on the UI thread, so unhandled exceptions will crash VS.
+            // Note: we don't expect any exceptions to be thrown, since the called method
+            // does all of its work on a background thread.
+            try
+            {
+                UpdateFileStorageAsync().Forget();
+            }
+            catch (Exception ex) when (!ErrorHandler.IsCriticalException(ex))
+            {
+                // Squash non-critical exceptions
+                logger.LogDebugExtended(ex.ToString());
+            }
         }
 
-        public void UpdateFileStorage()
+        /// <summary>
+        /// Updates the Roslyn suppressed issues file if in connected mode
+        /// </summary>
+        /// <remarks>The method will switch to a background if required, and will *not*
+        /// return to the UI thread on completion.</remarks>
+        public async Task UpdateFileStorageAsync()
         {
-            threadHandling.Run(async () =>
-            {
-                var sonarProjectKey = activeSolutionBoundTracker.CurrentConfiguration.Project?.ProjectKey;
+            logger.LogDebugExtended("Start");
```

Comment:
The new debug method will log the filename, caller, thread id and timestamp automatically.